docs/csharp/tour-of-csharp/program-structure.md

```diff
@@ -0,0 +1,106 @@
+---
+title: C# Program Structure | A Tour of the C# Language
+description: Learn the basic building blocks of a C# program
+keywords: .NET .NET Core
+author: BillWagner
+manager: wpickett
+ms.date: 08/10/2016
+ms.topic: article
+ms.prod: visual-studio-dev-14
+ms.technology: devlang-csharp
+ms.devlang: csharp
+ms.assetid: 984f0314-507f-47a0-af56-9011243f5e65
+---
+
+# Program Structure
+
+The key organizational concepts in C# are ***programs***, ***namespaces***, ***types***, ***members***, and ***assemblies***. C# programs consist of one or more source files. Programs declare types, which contain members and can be organized into namespaces. Classes and interfaces are examples of types. Fields, methods, properties, and events are examples of members. When C# programs are compiled, they are physically packaged into assemblies. Assemblies typically have the file extension `.exe` or `.dll`, depending on whether they implement ***applications*** or ***libraries***, respectively.
+
+The example
+
+```csharp
+using System;
+namespace Acme.Collections
+{
+    public class Stack
+    {
+        Entry top;
+        public void Push(object data) 
+		{
+			top = new Entry(top, data);
+		}
+		
+		public object Pop() 
+		{
+			if (top == null)
+			{
+				throw new InvalidOperationException();
+			}
+			object result = top.data;
+			top = top.next;
+			return result;
+		}
+
+		class Entry
+		{
+			public Entry next;
+			public object data;
+			public Entry(Entry next, object data) 
+			{
+				this.next = next;
+				this.data = data;
+			}
+		}
+	}
+}
+```
+
+declares a class named `Stack` in a namespace called `Acme.Collections`. The fully qualified name of this class is `Acme.Collections.Stack`. The class contains several members: a field named `top`, two methods named `Push` and `Pop`, and a nested class named `Entry`. The `Entry` class further contains three members: a field named `next`, a field named `data`, and a constructor. Assuming that the source code of the example is stored in the file `acme.cs`, the command line
```

Comment:
I find it odd to break the first sentence as this. Maybe move that first part above the example?